eng/scripts/Test-Code.ps1

```diff
@@ -73,67 +97,79 @@ function CopyNativeBinaryToTestDirs {
-        [string[]]$areas = @()
+        [string[]]$testsRootDirs
     )
     Write-Host "Copying native AzureMcp to test directories"
 
-    if (!$areas) {
-        $testDirectories = Get-ChildItem -Path $RepoRoot -Recurse -Filter "*.LiveTests" -Directory
-    } else {
-        $testDirectories = @()
-        foreach ($area in $areas) {
-            $areaName = $area.ToLower()
-            $areaPath = $areaName -eq 'core' ? "$RepoRoot/core/tests" : "$RepoRoot/areas/$areaName/tests"
-            if (Test-Path $areaPath) {
-                $areaTestDirectories = Get-ChildItem -Path $areaPath -Recurse -Filter "*.LiveTests" -Directory
-                $testDirectories += $areaTestDirectories
-            }
-        }
+    $testProjectDirs = @()
+    foreach ($testsRootDir in $testsRootDirs) {
+        $areaTestProjectDirs = Get-ChildItem -Path $testsRootDir -Recurse -Filter "*.LiveTests" -Directory
+        $testProjectDirs += $areaTestProjectDirs
     }
 
-    foreach ($testDir in $testDirectories) {
+    foreach ($testDir in $testProjectDirs) {
         $targetDirectory = "$($testDir.FullName)/bin/Debug/net9.0"
         Copy-Item $nativeBinaryPath $targetDirectory -Force
     }
 }
 
-function AddTestProjects($path) {
-    if($TestType -in @('Live', 'All')) {
-        $script:testProjects += Get-ChildItem $path -Recurse -File -Filter "*.LiveTests.csproj"
+function CreateTestSolution {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$workPath,
+        [Parameter(Mandatory=$true)]
+        [string[]]$testsRootDirs,
+        [Parameter(Mandatory=$true)]
+        [string]$testType
+    )
+
+    $testProjects = @()
+    foreach ($testsRootDir in $testsRootDirs) {
+        if($testType -in @('Live', 'All')) {
+            $testProjects += Get-ChildItem $testsRootDir -Recurse -File -Filter "*.LiveTests.csproj"
+        }
+        if($testType -in @('Unit', 'All')) {
+            $testProjects += Get-ChildItem $testsRootDir -Recurse -File -Filter "*.UnitTests.csproj"
+        }
     }
-    if($TestType -in @('Unit', 'All')) {
-        $script:testProjects += Get-ChildItem $path -Recurse -File -Filter "*.UnitTests.csproj"
+
+    if($testProjects.Count -eq 0) {
+        Write-Error "No test projects found in the specified areas for test type '$testType'."
+        return $false
     }
-}
 
-if (!$Areas) {
-    AddTestProjects $RepoRoot
-} else {
-    foreach ($area in $Areas) {
-        $areaName = $area.ToLower()
-        $areaPath = $areaName -eq 'core' ? "$RepoRoot/core/tests" : "$RepoRoot/areas/$areaName/tests"
-        if (Test-Path $areaPath) {
-            AddTestProjects $areaPath
-        } else {
-            Write-Error "Area path '$areaPath' does not exist."
-            return
-        }
+    # Create solution and add projects
+    Write-Host "Creating temporary solution file..."
+    Push-Location $workPath
+    try {
+        dotnet new sln -n "Tests" | Out-Null
+        dotnet sln add $testProjects --in-root
```

Comment:
IIRC, there is output from this command, should be piped to Out-Null... any output from an powershell function is interpreted as part of the result.